File: code/csharp/DesignPatterns/interpreter/InterpreterClient.cs
using System;

namespace Interpreter
{
    public class InterpreterClient
    {
        public static void InterpreterExample()
        {
            Context context = new Context("English");
            Interpreter interpreter = new Interpreter();

            string expression = "Hello";
            Console.WriteLine($"Client: Sending expression - {expression}");
            interpreter.Interpret(expression, context);

            string complexExpression = "How are you?";
            Console.WriteLine($"Client: Sending expression - {complexExpression}");
            interpreter.Interpret(complexExpression, context);
        }
    }
}

File: code/typescript/src/interpreter/InterpreterClient.ts
import { Interpreter } from "./Interpreter";
import { Context } from "./Context";

let context = new Context("English");
let interpreter = new Interpreter();

let expression = "Hello";
console.log(`Client: Sending expression - ${expression}`);
interpreter.interpret(expression, context);

let complexExpression = "How are you?";
console.log(`Client: Sending expression - ${complexExpression}`);
interpreter.interpret(complexExpression, context);
